### ISSUE TITLE
SNOW-297036: add description information for get_results_from_sfqid

### ISSUE THREAD

**Issue by javier5109** [snowflakedb/snowflake-connector-python#647]
## What is the current behavior?
newcur2.get_results_from_sfqid(query_id) only grabs result information and does not include description.

This is due to line 991 which excludes the description(link down below)  
 
self._result = self._inner_cursor._result
https://github.com/snowflakedb/snowflake-connector-python/blob/master/src/snowflake/connector/cursor.py

## What is the desired behavior?
add argument to not exclude description data

## How would this improve `snowflake-connector-python`?
this would follow conn.curose().execute logic where description data is automatically downloaded and would allow fetch_pandas_all() to work with the get_results_from_sfqid( query_id ) function

## References, Other Background
following this snowflake doc example, I cannot get description data for the query id
https://docs.snowflake.com/en/user-guide/python-connector-example.html#label-python-connector-asynchronous-query-examples

**Comment by dennis-weyland-by** [comment]
I also run in the same problem. To add a reproducible example:
```
snowflake_cursor.execute('SELECT * FROM "SNOWFLAKE"."ACCOUNT_USAGE"."USERS"')
df = snowflake_cursor.fetch_pandas_all()
```
will work while
```
snowflake_cursor.execute_async('SELECT * FROM "SNOWFLAKE"."ACCOUNT_USAGE"."USERS"')
snowflake_cursor.get_results_from_sfqid(snowflake_cursor.sfqid)
df = snowflake_cursor.fetch_pandas_all()
```
will result in the error:

>File "[...]/snowflake/connector/cursor.py", line 773, in fetch_pandas_all
>     if self._query_result_format != 'arrow':
> AttributeError: 'SnowflakeCursor' object has no attribute '_query_result_format'

**Comment by dennis-weyland-by** [comment]
Just found the `query_result` function from cursor. So basically this is working:
```
snowflake_cursor.execute_async('SELECT * FROM "SNOWFLAKE"."ACCOUNT_USAGE"."USERS"')
snowflake_cursor.query_result(snowflake_cursor.sfqid)
df = snowflake_cursor.fetch_pandas_all()
```
So the question is why `query_result` exists (I couldn't find it in the docs) and whether is safe to use this function instead of `get_results_from_sfqid`.

**Comment by javier5109** [comment]
@dennis-weyland-by  

The difference between query_result and .get_results_from_sfqid is that the latter will wait until the query is done running. The first will not. You can change the GITHUB code to return the results. Implementing it in the snowflake-connector code is trivial, You can just change the file code in the cursor.py file in your local machine to

def get_results_from_sfqid(self, sfqid: str, **result_flag = False**):
        """Gets the results from previously ran query."""

        def wait_until_ready():
            """Makes sure query has finished executing and once it has retrieves results."""
            no_data_counter = 0
            retry_pattern_pos = 0
            while True:
                status = self.connection.get_query_status(sfqid)
                if not self.connection.is_still_running(status):
                    break
                if status == QueryStatus.NO_DATA:  # pragma: no cover
                    no_data_counter += 1
                    if no_data_counter > ASYNC_NO_DATA_MAX_RETRY:
                        raise DatabaseError(
                            "Cannot retrieve data on the status of this query. No information returned "
                            "from server for query '{}'"
                        )
                time.sleep(
                    0.5 * ASYNC_RETRY_PATTERN[retry_pattern_pos]
                )  # Same wait as JDBC
                # If we can advance in ASYNC_RETRY_PATTERN then do so
                if retry_pattern_pos < (len(ASYNC_RETRY_PATTERN) - 1):
                    retry_pattern_pos += 1
            if status != QueryStatus.SUCCESS:
                raise DatabaseError(
                    "Status of query '{}' is {}, results are unavailable".format(
                        sfqid, status.name
                    )
                )
            self._inner_cursor.execute(
                "select * from table(result_scan('{}'))".format(sfqid)
            )
            **if result_flag:
                   self._result = self._inner_cursor._result
            else:
                  sel.result = self.inner_cursor**
            # Unset this function, so that we don't block anymore
            self._prefetch_hook = None

        self.connection.get_query_status_throw_if_error(
            sfqid
        )  # Trigger an exception if query failed
        klass = self.__class__
        self._inner_cursor = klass(self.connection)
        self._sfqid = sfqid
        self._prefetch_hook = wait_until_ready



I am not allowed to change the packages at work, so I came up with my own implementation. I rather not do this as I would have to maintain this and its not as reliable as what the snowflake team has developed. 

def snowflake_download_query_id(query_id):
    ctx = snowflake.connector.connect()
    cs = ctx.cursor()
    
    #cs.get_results_from_sfqid(query_id)
    #wait 
    STATUS = ctx.get_query_status(query_id)
    RUNNING = ctx.is_still_running(STATUS)
    

    while True:
        if RUNNING == False:
            print('-->Query is done running, downloading query_id {}'.format(query_id))
            break
        else:
            print("-->Query {} is still running. Sleeping for 2 minutes ".format(query_id))
            time.sleep(60*2)
            
            RUNNING = ctx.is_still_running(STATUS)
            STATUS = ctx.get_query_status(query_id)
            print(STATUS)
    

    start = time.time()
    cs.execute('select * from table(result_scan(\'{}\'))'.format(query_id))
    print("-->finished executing query in in {} minutes".format((time.time() - start)/60))
    start = time.time()
    
    #df = cs.fetch_pandas_all()
    df = pd.concat([x for x in cs.fetch_pandas_batches()], ignore_index = True, sort = False)

    print(count)
    print("-->finished downloading query in in {} minutes".format((time.time() - start)/60))

    ctx.close()
    return df

**Comment by info-rchitect** [comment]
I'll add my voice that getting `fetch_pandas_all` working with `execute_async` is important :)

**Comment by sfc-gh-mkeller** [comment]
@dennis-wey 's PR was merged through #737
